lib/renderer/ipc-renderer-internal-utils.ts

```diff
@@ -1,6 +1,21 @@
 import { ipcRendererInternal } from '@electron/internal/renderer/ipc-renderer-internal'
 import * as errorUtils from '@electron/internal/common/error-utils'
 
+type IPCHandler = (event: Electron.IpcRendererEvent, ...args: any[]) => any
+
+export const handle = function <T extends IPCHandler> (channel: string, handler: T) {
+  ipcRendererInternal.on(channel, (event, requestId, ...args) => {
+    new Promise(resolve => resolve(handler(event, ...args))
```

Comment:
```js
Promise.resolve(handler(...))
```

Reply:
this would not return a rejected promise if handler throws